hudi-spark-datasource/hudi-spark3.3.x/src/main/antlr4/imports/SqlBase.g4

```diff
@@ -0,0 +1,1908 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file is an adaptation of Presto's presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4 grammar.
+ */
+
+// The parser file is forked from spark 3.2.0's SqlBase.g4.
```

Comment:
Need copy spark 3.3's `SqlBaseParser.g4`?

Reply:
Those .g4 files have been refactored and changed a lot in Spark 3.3. e.g.: https://github.com/apache/spark/pull/35701
And I don't think it's needed to port those changes back to Hudi as they are going to be removed soon.

Reply:
Is this going to be compatible with Spark 3.3 and support new SQL features?

Reply:
It's gonna be compatible but new SQL feature might not be supported. 

This should be solved after simplifying time travel logic, jira: https://issues.apache.org/jira/browse/HUDI-4468